tests/hdfs_kerberos_test.go

```diff
@@ -0,0 +1,150 @@
+	hdfsKerberosDeployment = []Resource{
+		{
+			name: "configmaps",
+			path: "configmaps",
+		},
+		{
+			name: "volumes",
+			path: "volumes",
+		},
+		{
+			name: "kerberos",
+			path: "kerberos-deployment.yaml",
+			wait: true,
+		},
+		{
+			name: "hdfs-namenode",
+			path: "namenode-deployment.yaml",
+			wait: true,
+		},
+		{
+			name: "hdfs-datanode",
+			path: "datanode-deployment.yaml",
+			wait: true,
+		},
+	}
+	eventLogDir       = "hdfs://namenode.hdfs-kerberos.svc.cluster.local:9000/history"
+	hadoopTokenSecret = "hadoop-token"
+)
+
+type Resource struct {
+	name string
+	path string
+	wait bool
+}
+
+type HdfsIntegrationSuite struct {
+	operator utils.SparkOperatorInstallation
+	suite.Suite
+}
+
+func TestHdfsIntegrationSuite(t *testing.T) {
+	suite.Run(t, new(HdfsIntegrationSuite))
+}
+
+func (suite *HdfsIntegrationSuite) SetupSuite() {
+	utils.Kubectl("create", "ns", namespace)
+	// deploy KDC and HDFS
+	for _, resource := range hdfsKerberosDeployment {
+		utils.Kubectl("apply", "-f", fmt.Sprint(resourceFolder, "/", resource.path), "-n", namespace)
+
+		if resource.wait {
+			utils.Kubectl("wait", fmt.Sprint("deployments/", resource.name),
+				"--for=condition=available", "--timeout=3m", "-n", namespace)
+		}
+	}
+	nameNodePod, err := utils.Kubectl("get", "pods", "--selector=name=hdfs-namenode",
+		"--output=jsonpath={.items[*].metadata.name}", "-n", namespace)
+	if err != nil {
+		suite.FailNow("Error while getting a name of Namenode pod", err)
+	}
+	// run init script to copy test data to HDFS and export delegation token
+	utils.Kubectl("exec", nameNodePod, "-n", namespace, "--", "init.sh")
+
+	// copy delegation token from the pod to a local filesystem
+	utils.Kubectl("cp", fmt.Sprint(nameNodePod, ":", "tmp/hadoop.token"), "/tmp/hadoop.token", "-n", namespace)
+	utils.Kubectl("create", "secret", "generic", hadoopTokenSecret, "--from-file", "/tmp/hadoop.token", "-n", namespace)
+}
+
+// invoked before each test
+func (suite *HdfsIntegrationSuite) BeforeTest(suiteName, testName string) {
+	fmt.Println(suiteName, testName)
+
+	suite.operator = utils.SparkOperatorInstallation{
+		Namespace:            namespace,
+		SkipNamespaceCleanUp: true,
+	}
+	if testName == "Test_Spark_Hdfs_Kerberos_SparkHistoryServer" {
+		operatorParams := map[string]string{
+			"enableHistoryServer":         "true",
+			"historyServerFsLogDirectory": eventLogDir,
+			"delegationTokenSecret":       hadoopTokenSecret,
+		}
+		suite.operator.Params = operatorParams
+	}
+	if err := suite.operator.InstallSparkOperator(); err != nil {
+		suite.FailNow(err.Error())
+	}
+}
+
+func (suite *HdfsIntegrationSuite) AfterTest(suiteName, testName string) {
+	utils.Kubectl("kudo", "uninstall",
+		"--instance", suite.operator.InstanceName,
+		"--namespace", suite.operator.Namespace)
+	// TODO: temporary workaround: explicitly waiting for uninstall to clean all the resources before the next test starts.
+	time.Sleep(15 * time.Second)
```

Comment:
can this workaround be replaced with periodic polling with timeout?

Reply:
The issue with `kudo uninstall` is that we can't actually verify all the related resources are deleted. What I observed locally, is sometimes the second test failed because of the incomplete cleanup after `uninstall`, meaning when the next installation starts, some resources of the previous one are still not cleaned up, whick cause deploy plan failure.
So, as it was suggested in [Slack thread](https://mesosphere.slack.com/archives/CR1SVH3PD/p1582186988020600), one of the possible solution is to delete the instance itself and together with `--wait` flag.

Reply:
Looking at the failed build history, I would say a solution with the `wait` flag could not be considered reliable. I going to switch back to the default `operator.Cleanup()` approach, but I need to make some changes and separate the Operator and HDFS with Kerberos using different namespaces. This will allow us to clean operators and preserve HDFS installation.